zbus_macros/src/iface.rs

```diff
@@ -1051,37 +1053,38 @@ fn get_args_from_inputs(
 
                 let signal_context_arg = &input.pat;
 
-                signal_emitter_arg_decl = Some(quote! {
-                    let #signal_context_arg = match hdr.path() {
-                        ::std::option::Option::Some(p) => {
-                            #zbus::object_server::SignalEmitter::new(connection, p).expect("Infallible conversion failed")
-                        }
-                        ::std::option::Option::None => {
-                            let err = #zbus::fdo::Error::UnknownObject("Path Required".into());
-                            return connection.reply_dbus_error(&hdr, err).await;
-                        }
-                    };
-                });
+                signal_emitter_arg_decl = match method_type {
+                    MethodType::Property(PropertyType::Getter) => {
+                        Some(quote! { let #signal_context_arg = signal_emitter.clone(); })
```

Comment:
Full path in generated code please: `::std::Clone::clone(signal_emitter)`. Don't feel bad about it though. I make that mistake too. It's easy to forget.

Reply:
Huh. I thought that in this particular case, since `signal_emitter` is accessible and usable, that the Rust compiler would be able to figure out correctly which `.clone()` we mean, but I admit that's "based" more on the rules of dependent lookup in C++ that I'm familiar with, than any actual Rust documentation. :slightly_smiling_face: 

I will of course update this. Thanks!

Reply:
> I thought that in this particular case, since `signal_emitter` is accessible and usable, that the Rust compiler would be able to figure out correctly which `.clone()` we mean

You're right. In this specific case, I see very little possibility of it being ambiguous in any context but there is no way to be sure about it. In general it's best to just always be explicit in the generated code.